src/pip/_internal/commands/__init__.py

```diff
@@ -18,7 +18,7 @@
 # in a test-related module).
 #    Finally, we need to pass an iterable of pairs here rather than a dict
 # so that the ordering won't be lost when using Python 2.7.
-commands_dict = OrderedDict([
+commands_dict: "OrderedDict[str, CommandInfo]" = OrderedDict([
```

Comment:
You need to jump through some hoops for this to work on 3.6.

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import OrderedDict
else:
    from collections import OrderedDict

commands_dict: "OrderedDict[str, CommandInfo]" = OrderedDict(...)
```

But does this has to be an ordered dict? Usually it's good enough to just annotate this as dict.

Reply:
> But does this have to be an ordered dict? Usually it's good enough to just annotate this as dict.

Agree, I'll change `OrderedDict` to `Dict`.

Reply:
This is what you meant, right?

https://github.com/pypa/pip/pull/10182/commits/10b35b61bd0ad7a7e01627262d16c75cedfe5ca5

Reply:
Yup 👍

Reply:
FWIW, given that we're Python 3.6+, we can just change this into a dictionary; since those are ordered starting Python 3.6.

Reply:
Sounds good, thanks for the suggestion!